react-native-reanimated.d.ts

```diff
@@ -380,8 +424,63 @@ declare module 'react-native-reanimated' {
       initialValue: T
     ): AnimatedValue<T>;
 
+    // reanimated2 functions
+    export function runOnUI<A, R>(fn: (...args: A) => R): (...args: Parameters<typeof fn>) => void;
+    export function processColor(color: number | string): number;
+                        
+    // reanimated2 hooks
+    export function useSharedValue<T extends SharedValueType>(
+      initialValue: T
+    ): SharedValue<T>;
+    export function useDerivedValue<T extends SharedValueType>(
+      processor: () => T
+    ): SharedValue<T>;
+    export function useAnimatedStyle<T extends ViewStyle, TextStyle, ImageStyle>(
```

Comment:
this works way better for me

```suggestion
    export function useAnimatedStyle<T extends AnimateStyle>(
```

Reply:
I’m not sure about this. `AnimateStyle` needs a mandatory generic, as you can see below.
https://github.com/software-mansion/react-native-reanimated/blob/fffcb9f541dd2d2009f8bb68283e81b62d16870c/react-native-reanimated.d.ts#L167